pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -134,21 +134,28 @@ def work(self):
                                             pokemon_potential
                                         ), 'green'
                                     )
-                                    if self.should_release_pokemon(pokemon_name, cp, pokemon_potential, response_dict):
-                                        id_list2 = self.count_pokemon_inventory()
-                                        # Transfering Pokemon
-                                        pokemon_to_transfer = list(
-                                            Set(id_list2) - Set(id_list1))
-                                        if len(pokemon_to_transfer) == 0:
-                                            raise RuntimeError(
-                                                'Trying to transfer 0 pokemons!')
-                                        self.transfer_pokemon(
-                                            pokemon_to_transfer[0])
-                                        logger.log(
-                                            '[#] {} has been exchanged for candy!'.format(pokemon_name), 'green')
+                                    pokemon_to_transfer = list(Set(id_list2) - Set(id_list1))
+                                    self.api.evolve_pokemon(pokemon_id=pokemon_to_transfer[0])
```

Comment:
Does it make sense to pull this out into a new worker called EvolvePokemonWorker? It would make this significantly easier to refactor later.